WORKSPACE

```diff
@@ -1,54 +1,26 @@
 workspace(name = "polaris")
 
-load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
-load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
-
-#----------- Cross-Compilation Support ---------------
+#-------------------------------------------------------------------------------
+# Point One cross-compilation support
+#
+# See https://github.com/curtismuntz/bazel_compilers/tree/boost-example.
 load("//compilers:dependencies.bzl", "cross_compiler_dependencies")
+
 cross_compiler_dependencies()
 
-#----------- gflags ---------------
-http_archive(
-    name = "com_github_gflags_gflags",
-    sha256 = "34af2f15cf7367513b352bdcd2493ab14ce43692d2dcd9dfc499492966c64dcf",
-    strip_prefix = "gflags-2.2.2",
-    urls = ["https://github.com/gflags/gflags/archive/v2.2.2.tar.gz"],
-)
-
-bind(
-    name = "gflags",
-    actual = "@com_github_gflags_gflags//:gflags",
-)
-
-#----------- glog ---------------
-git_repository(
-    name = "com_google_glog",
-    commit = "781096619d3dd368cfebd33889e417a168493ce7",
-    remote = "https://github.com/google/glog.git",
-    shallow_since = "1542766478 +0900",
-)
-
-#---------------ssl --------------
-git_repository(
-    name = "boringssl",
-    commit = "87f3087d6343b89142d1191388a5885d74459df2",
-    remote = "https://boringssl.googlesource.com/boringssl",
-    shallow_since = "1586306564 +0000",
-)
-
-bind(
-    name = "glog",
-    actual = "@com_google_glog//:glog",
-)
-
-#----------- boost ---------------
-git_repository(
-    name = "com_github_nelhage_rules_boost",
-    commit = "9f9fb8b2f0213989247c9d5c0e814a8451d18d7f",
-    remote = "https://github.com/nelhage/rules_boost",
-    shallow_since = "1570056263 -0700",
-)
-
-load("@com_github_nelhage_rules_boost//:boost/boost.bzl", "boost_deps")
-
-boost_deps()
+#-------------------------------------------------------------------------------
+# Import Polaris Client dependencies.
+load("//bazel:repositories.bzl", polaris_dependencies = "dependencies")
+
+polaris_dependencies()
+
+#-------------------------------------------------------------------------------
+# Import dependencies for example applications (not needed by the library
+# itself).
+load("//bazel:examples_repositories.bzl", polaris_examples_dependencies = "dependencies")
+
+polaris_examples_dependencies()
+
+load("//bazel:examples_dependencies.bzl", load_polaris_examples_dependencies = "load_dependencies")
```

Comment:
why the different name for the deps?

Reply:
You mean `dependencies()` vs `load_dependencies()`?

The first function (`dependencies()`) is the function that actually downloads the dependencies to create `git_repository` entries or whatever in the workspace. The second function calls the dependency functions for those third-party dependencies. In this case, it calls `boost_deps()`. Ideally, I'd like to just do that in one function. In order to call `boost_deps()` you have to first `load()` it though, and `load()` calls must be the first line in a bazel/workspace file so you can't do it after the `git_repostiory()` call, which needs to come first to define the thing to be loaded.

Wasn't sure of the best name for the second function. `dependencies()` for the first one is consistent with some other Bazel-supporting repos so I used that but I'm open to other names. The second one is a less common thing. I don't really love "load", just threw something in there to get it working but definitely open to changing it. Or alternatively call the first one "download" and the second one "load" or something?